### ISSUE TITLE
Rule "truthy" should not be upset about True vs true

### ISSUE THREAD

**Issue by IronTooch** [ansible-community/ansible-lint#1903]
##### Summary
The "truthy" rule provides a warning for using **True** vs **true**. truthy should allow both. 

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console
ansible --version
ansible [core 2.12.1]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/tooch/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/tooch/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True
ansible-lint --version
ansible-lint 5.3.2 using ansible 2.12.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Ubuntu 20 (Focal Fossa)

##### STEPS TO REPRODUCE

```yaml
---
- hosts: all
  tasks:
  - name: Make Super Secret Password
    set_fact:
      super_secret_pwd: "{{ lookup('community.general.random_string', length=24, special=false) }}"
    no_log: True

  - name: Set Super Secret Password config file
    ansible.builtin.copy:
      content: "{{ super_secret_password }}"
      dest: "/root/ss_password.txt"
      force: true
      owner: "root"
      group: "root"
      mode: "0700"
    no_log: True
```
```console
ansible-lint sample_playbook.yml
```

##### Desired Behavior

Using `True` should not result in any Linting warnings

##### Actual Behavior
```console
yaml: truthy value should be one of [false, true] (truthy)
sample_playbook.yml:7

yaml: truthy value should be one of [false, true] (truthy)
sample_playbook.yml:17
```

**Comment by IronTooch** [comment]
@cognifloyd That makes sense, I think we should add something to the Discussion board then to explain why the rule is implemented the way it is. I searched and couldn't find anything.

**Comment by cognifloyd** [comment]
I don't know why it was originally implemented like that. I gave my reasoning for why I wouldn't change it.

**Comment by IronTooch** [comment]
Forward compatibility is a great reason. This is just the first place where I would learn that, because it's not otherwise clear elsewhere.
